paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/lookup/FileStoreLookupFunction.java

```diff
@@ -206,7 +200,7 @@ private void open() throws Exception {
             partitionLoader.checkRefresh();
             List<BinaryRow> partitions = partitionLoader.partitions();
             if (!partitions.isEmpty()) {
-                lookupTable.specificPartitionFilter(createSpecificPartFilter(partitions));
+                lookupTable.specificPartitionFilter(partitionLoader.createSpecificPartFilter());
             }
         }
 
```

Comment:
Suggestion > Add a log where LookupTable is created:  Which type of LookupTable will be created and why..